src/pkgcheck/checks/eclass.py

```diff
@@ -30,6 +30,40 @@ def desc(self):
         return f'uses deprecated eclass: {self.eclass} ({replacement})'
 
 
+class DeprecatedEclassVariable(results.VersionResult, results.Warning):
+    """Package uses a deprecated variable from an eclass."""
+
+    def __init__(self, variable, replacement, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+        self.replacement = replacement
+
+    @property
+    def desc(self):
+        if self.replacement is not True:
```

Comment:
We're using a different pattern here than above (`is not None` vs `is not True`). Why not just `if self.replacement` always?

Reply:
We have 3 states:

1. `None` - not deprecated
2. `True` - no replacement string
3. other - the replacement string

I'm not sure what is the tag for eclass and why was it correct in that check. If I use `is not None` in my code, I get
```
DeprecatedEclassFunction: version 1.13.0: uses deprecated function: go-module_set_globals (migrate to True)
```

Reply:
Huh. OK, whatever. Thanks!

Reply:
A better contextual explanation (off the top of my head but probably still relevant): pkgcore eclassdoc parsing provides three states:

1. False - not deprecated (the default value)
2. True - deprecated with no replacement (as specified in the eclass if the replacement value is 'None' or 'none')
3. string value - deprecated with a given value as the replacement

pkgcheck maps this into two states for the deprecated eclass result (as it ignores the non-deprecated state):

1. None - deprecated with no replacement
2. string value - deprecated with a given value as a replacement

What should be done for all related results is what sam said (and probably was just overlooked during writing/refactoring eclassdoc parsing initially). To do that for this check just copy what is done for eclasses and map non-string values (True) to None for the replacement argument.